Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,302 @@
+                        sb.append(
+                                syn.getID().toString() + edu.asu.conceptpower.servlet.core.Constants.SYNONYM_SEPARATOR);
+                }
+                doc.add(new StringField(LuceneFieldNames.SYNONYMID, sb.toString(), Field.Store.YES));
+                // Adding this new data to delete only wordnet concepts while
+                // adding all wordnet concepts from jwi.
+                doc.add(new StringField(LuceneFieldNames.CONCEPTTYPE, "wordnetconcept", Field.Store.YES));
+                writer.addDocument(doc);
+            }
+
+        }
+    }
+
+    @Override
+    public void indexConcepts() throws LuceneException {
+
+        try {
+            String wnhome = configuration.getWordnetPath();
+            String path = wnhome + File.separator + configuration.getDictFolder();
+
+            URL url = null;
+
+            url = new URL("file", null, path);
+
+            IDictionary dict = new Dictionary(url);
+            dict.open();
+
+            // 2. Adding data into
+            Iterator<IIndexWord> iterator = dict.getIndexWordIterator(POS.NOUN);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.ADVERB);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.ADJECTIVE);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.VERB);
+            createDocuments(iterator, dict, writer);
+
+            writer.commit();
+
+        } catch (IOException ex) {
+            throw new LuceneException("Problem in Creating Index. Please retry", ex);
+        }
+    }
+
+    public ConceptEntry[] queryIndex(Map<String, String> fieldMap, String operator) throws LuceneException, IllegalAccessException {
+
+        if (operator == null) {
+            operator = "AND";
+        }
+
+        StringBuffer queryString = new StringBuffer();
+        int firstEntry = 1;
+
+        ConceptEntry con = new ConceptEntry();
+        java.lang.reflect.Field[] fields = con.getClass().getDeclaredFields();
+        for (java.lang.reflect.Field field : fields) {
+            SearchField search = field.getAnnotation(SearchField.class);
+            LuceneField luceneFieldAnnotation = field.getAnnotation(LuceneField.class);
+            if (search != null) {
+                String searchString = fieldMap.get(search.fieldName());
+                if (searchString != null) {
+                    if (firstEntry != 1)
+                        queryString.append(" " + operator + " ");
+                    firstEntry++;
+                    queryString.append(luceneFieldAnnotation.lucenefieldName() + ":");
+                    queryString.append(searchString);
+                }
+            }
+        }
+
+        List<ConceptEntry> concepts = new ArrayList<ConceptEntry>();
+        int hitsPerPage = 10;
+        IndexSearcher searcher = null;
```

Comment:
This should be a class variable. From the documentation:
"For performance reasons, if your index is unchanging, you should share a single IndexSearcher instance across multiple searches instead of creating a new one per-search. If your index has changed and you wish to see the changes reflected in searching, you should use DirectoryReader.openIfChanged(DirectoryReader) to obtain a new reader and then create a new IndexSearcher from that."